### ISSUE TITLE
Feature request: Return stale value while revalidating

### ISSUE THREAD

**Issue by futpib** [sindresorhus/p-memoize#19]
When used with `maxAge` option, when the cache entry has expired, memoized function could return last cached value while waiting for a new result to arrive.

**Comment by papb** [comment]
I think this could be an option, but not on by default.

**Comment by sindresorhus** [comment]
Yeah, sounds useful as an opt-in option.

**Comment by AmrSaber** [comment]
Hello @sindresorhus, I am interested in helping out with this feature. How can I help with it?

**Comment by Richienb** [comment]
@AmrSaber PRs are always welcome

**Comment by AmrSaber** [comment]
I don't understand how #26 closes this ticket? How do we achieve the stale-while-revalidate scenario now?

**Comment by Richienb** [comment]
#26 Moves this feature into userland.

**Comment by AmrSaber** [comment]
Can you show an example of how to achieve this?

**Comment by Richienb** [comment]
False alarm: after the previous PR, I'm not sure how to check if a value is *stale* anymore to then call the function again.

**Comment by futpib** [comment]
Should this be reopened then?

**Comment by Richienb** [comment]
Yes, but with the new goal of adding an API to the cache storage object to check whether the object is stale.
